rewrite-java/src/main/java/org/openrewrite/java/style/ImportLayoutStyle.java

```diff
@@ -788,11 +788,8 @@ private static String packageOrOuterClassName(JRightPadded<J.Import> anImport) {
         if (anImport.getElement().isStatic()) {
             return typeName;
         } else {
-            String className = anImport.getElement().getClassName();
-            if (className.contains("$")) {
-                return anImport.getElement().getPackageName() + "." +
-                        className.substring(0, className.lastIndexOf('$'))
-                                .replace('$', '.');
+            if (typeName.contains("$")) {
+                return typeName.substring(0, typeName.lastIndexOf('$'));
```

Comment:
Was this a necessary change, or just a simplification you'd thought to apply?
The way it's written I wonder if it had meant to handled twice-nested inner classes.

Reply:
This was a necessary change as it never reached inside the if because `anImport.getElement().getClassName();` returns an already sanitized reference, you were right about having to handle multiple inner classes though, So I've updated it